src/ae/contract.ts

```diff
@@ -0,0 +1,212 @@
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * Contract module - routines to interact with the æternity contract
+ *
+ * High level documentation of the contracts are available at
+ * https://github.com/aeternity/protocol/tree/master/contracts and
+ *
+ * @module @aeternity/aepp-sdk/es/ae/contract
+ * @export Contract
+ * @example import { Contract } from '@aeternity/aepp-sdk'
+ */
+
+// @ts-expect-error TODO remove
+import Ae from '.'
+import ContractCompilerHttp from '../contract/compiler'
+import getContractInstance from '../contract/aci'
+// @ts-expect-error TODO remove
+import { AMOUNT, MIN_GAS_PRICE } from '../tx/builder/schema'
+// @ts-expect-error TODO remove
+import { decode, produceNameId } from '../tx/builder/helpers'
+// @ts-expect-error TODO remove
+import { getNetworkId } from '../node'
+
+export abstract class _Contract {
+  /**
+   * Utility method to create a delegate signature for a contract
+   * @alias module:@aeternity/aepp-sdk/es/ae/contract
+   * @category async
+   * @param ids The list of id's to prepend
+   * @param opt Options
+   * @return Signature in hex representation
+   */
+  async delegateSignatureCommon (
+    ids: string[] = [],
+    opt: { onAccount?: string | object } = {}): Promise<string> {
+    const signature = await this.sign(
+      Buffer.concat([
+        Buffer.from(this.getNetworkId(opt)),
+        ...ids.map(e => decode(e))
+      ]),
+      opt
+    )
+    return Buffer.from(signature).toString('hex')
+  }
+
+  /**
+   * Helper to generate a signature to delegate pre-claim/claim/transfer/revoke of a name to
+   * a contract.
+   * @alias module:@aeternity/aepp-sdk/es/ae/contract
+   * @category async
+  *  @param args
+   * @param args.contractId Contract Id
+   * @param args.name The name
+   * @param opt Options
+   * @returns Signature for delegation
+   * @example
+   * const aeSdk = await Universal({ ... })
+   * const contractId = 'ct_asd2ks...' // contract address
+   * const name = 'example.chain' // AENS name
+   * const onAccount = await aeSdk.address() // Sign with a specific account
+   * // Preclaim signature
+   * const params = { contractId }
+   * const preclaimSig = await aeSdk.createAensDelegationSignature(params, { onAccount: current })
+   * // Claim, transfer and revoke signature
+   * const params = { contractId, name }
+   * const aensDelegationSig = await contract.createAensDelegationSignature(
+   *   params, name, { onAccount: current }
+   * )
+   */
+  async createAensDelegationSignature (
+    { contractId, name }: {contractId: string, name?: string[]},
```

Comment:
```suggestion
    { contractId, name }: {contractId: string, name?: string},
```
there is a single name